yal/main.py

```diff
@@ -855,3 +861,11 @@ async def publish_message(self, question):
         """
         content = replace_persona_fields(question, self.user.metadata)
         await self.worker.publish_message(self.inbound.reply(content))
+
+    async def state_survey_validation(self):
+        """
```

Comment:
Is `main.py` the right place for this state? It's very survey specific so maybe it should live there?